glean-core/ios/Glean/Metrics/TimingDistributionMetric.swift

```diff
@@ -0,0 +1,152 @@
+    /// count to the corresponding bucket in the timing distribution.
+    /// This will record an error if no `start()` was called.
+    ///
+    /// - parameters:
+    ///     * timerId: The `GleanTimerId` to associate with this timing.
+    ///                This allows for concurrent timing of events associated with different ids
+    ///                to the same timespan metric.
+    public func stopAndAccumulate(_ timerId: GleanTimerId?) {
+        // `start` might return null.
+        // Accepting that means users of this API don't need to do a null check.
+        guard !self.disabled else { return }
+        guard let timerId = timerId else { return }
+
+        // The Rust code runs async and might be delayed. We need the time as precisely as possible.
+        // We also need the same clock for start and stop (`start` takes the time on the Swift side).
+        let stopTime = timestampNanos()
+
+        Dispatchers.shared.launchAPI {
+            glean_timing_distribution_set_stop_and_accumulate(
+                Glean.shared.handle,
+                self.handle,
+                timerId,
+                stopTime
+            )
+        }
+    }
+
+    /// Abort a previous `start()` call. No error is recorded if no `start()` was called.
+    ///
+    /// - parameters:
+    ///     * timerId: The `GleanTimerId` to associate with this timing.
+    ///                This allows for concurrent timing of events associated with different ids
+    ///                to the same timing distribution metric.
+    public func cancel(_ timerId: GleanTimerId?) {
+        guard !self.disabled else { return }
+        guard let timerId = timerId else { return }
+
+        Dispatchers.shared.launchAPI {
+            glean_timing_distribution_cancel(self.handle, timerId)
+        }
+    }
+
+    /// Tests whether a value is stored for the metric for testing purposes only. This function will
+    /// attempt to await the last task (if any) writing to the the metric's storage engine before
+    /// returning a value.
+    ///
+    /// - parameters:
+    ///     * pingName: represents the name of the ping to retrieve the metric for.
+    ///                 Defaults to the first value in `sendInPings`.
+    /// - returns: true if metric value exists, otherwise false.
+    func testHasValue(_ pingName: String? = nil) -> Bool {
+        Dispatchers.shared.assertInTestingMode()
+
+        let pingName = pingName ?? self.sendInPings[0]
+        return glean_timing_distribution_test_has_value(Glean.shared.handle, self.handle, pingName).toBool()
+    }
+
+    /// Returns the stored value for testing purposes only. This function will attempt to await the
+    /// last task (if any) writing to the the metric's storage engine before returning a value.
+    ///
+    /// Throws a "Missing value" exception if no value is stored
+    ///
+    /// -parameters:
+    ///   * pingName: represents the name of the ping to retrieve the metric for.
```

Comment:
super nit: indentation of list item doesn't match other function parameter docs, looks like it's missing a space?